integration-test/src/test/java/org/jboss/pnc/integration/ProductReleaseRestTest.java

```diff
@@ -0,0 +1,176 @@
+        productMilestoneTemplate.addValue("_productVersionId", String.valueOf(productVersionId));
+        Response response = given().header("Accept", "application/json").header("Authorization", "Bearer " + authProvider.getTokenString())
+                .body(productMilestoneTemplate.fillTemplate()).contentType(ContentType.JSON).port(getHttpPort()).when()
+                .post(PRODUCT_MILESTONE_REST_ENDPOINT);
+        Assertions.assertThat(response.statusCode()).isEqualTo(201);
+        String location = response.getHeader("Location");
+        productMilestoneId = Integer.valueOf(location.substring(location.lastIndexOf(
+                PRODUCT_MILESTONE_REST_ENDPOINT) + PRODUCT_MILESTONE_REST_ENDPOINT.length()));
+    }
+
+    @Test
+    @InSequence(2)
+    public void prepareProductReleaseId() {
+        given().header("Accept", "application/json").header("Authorization", "Bearer " + authProvider.getTokenString())
+                    .contentType(ContentType.JSON).port(getHttpPort()).when()
+                .get(String.format(PRODUCT_RELEASE_REST_ENDPOINT)).then().statusCode(200)
+                .body(JsonMatcher.containsJsonAttribute("[0].id", value -> productReleaseId = Integer.valueOf(value)));
+    }
+
+    @Test
+    @InSequence(3)
+    public void shouldGetSpecificProductRelease() {
+        given().header("Accept", "application/json").header("Authorization", "Bearer " + authProvider.getTokenString())
+                .contentType(ContentType.JSON).port(getHttpPort()).when()
+                .get(String.format(PRODUCT_RELEASE_SPECIFIC_REST_ENDPOINT, productReleaseId)).then().statusCode(200)
+                .body(JsonMatcher.containsJsonAttribute("id"));
+    }
+
+    @Test
+    @InSequence(4)
+    public void shouldCreateNewProductRelease() throws IOException {
+        JsonTemplateBuilder productReleaseTemplate = JsonTemplateBuilder.fromResource("productRelease_template");
+        productReleaseTemplate.addValue("_productVersionId", String.valueOf(productVersionId));
+        productReleaseTemplate.addValue("_productMilestoneId", String.valueOf(productMilestoneId));
+
+        Response response = given().header("Accept", "application/json").header("Authorization", "Bearer " + authProvider.getTokenString())
+                .body(productReleaseTemplate.fillTemplate()).contentType(ContentType.JSON).port(getHttpPort()).when()
+                .post(PRODUCT_RELEASE_REST_ENDPOINT);
+        Assertions.assertThat(response.statusCode()).isEqualTo(201);
+
+        String location = response.getHeader("Location");
+        logger.info("Found location in Response header: " + location);
+
+        newProductReleaseId = Integer.valueOf(location.substring(location.lastIndexOf(
+                PRODUCT_RELEASE_REST_ENDPOINT) + PRODUCT_RELEASE_REST_ENDPOINT.length()));
+
+        logger.info("Created id of product version: " + newProductReleaseId);
```

Comment:
The important assertion is on line 124.  The remaining code is just parsing the ID from the URL and logging.  I guess we could assert that a valid Integer is found in the URL?